mikeio/generic.py

```diff
@@ -552,6 +552,82 @@ def concat(
                 )  # get end time from current file
                 dfs_i.Close()
 
+        if keep == "average":
+            if i == 0:
+                # For first file, write all timesteps normally

```

Comment:
Isn't it better to calculate the timestep where the overlap occurs, calcuate the average, write that and then the rest, instead of writing the entire contents and then trying to overwrite?

Reply:
Agree. I would suggest an algorithm like this (file1 starts before file2): 

* open file1
* write file1 timesteps until start of file file2
* open file2
* write overlapping steps by with alpha*data1 + (1-alpha)*data2 until end of file1
* close file1
* write rest of file2 
* close file2 

using the alpha notation above it will be super simple to later add a keep="linear" that linear mixes the two in the overlapping period.

Reply:
don't shoot the messenger, I chat'gpted this solution due to my lack of mikecore competences